pypeit/specobjs.py

```diff
@@ -387,34 +143,39 @@ def get_std(self):
 
         """
         # Is this MultiSlit or Echelle
-        pypeline = (self.pypeline)[0]
+        pypeline = (self.PYPELINE)[0]
         if 'MultiSlit' in pypeline:
-            nspec = self[0].optimal['COUNTS'].size
             SNR = np.zeros(self.nobj)
             # Have to do a loop to extract the counts for all objects
             for iobj in range(self.nobj):
-                SNR[iobj] = np.median(self[iobj].optimal['COUNTS']*np.sqrt(self[iobj].optimal['COUNTS_IVAR']))
+                SNR[iobj] = np.median(self[iobj].OPT_COUNTS*np.sqrt(
+                    self[iobj].OPT_COUNTS_IVAR))
+            # Maximize S/N
             istd = SNR.argmax()
+            # Return
             return SpecObjs(specobjs=[self[istd]])
         elif 'Echelle' in pypeline:
-            uni_objid = np.unique(self.ech_objid)
-            uni_order = np.unique(self.ech_orderindx)
+            uni_objid = np.unique(self.ECH_FRACPOS)  # A little risky using floats
```

Comment:
Isn't ech_objid still an attribute.

Reply:
I don't use it for object naming any longer.